Composer/packages/client/src/telemetry/useInitializeLogger.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { useCallback, useEffect } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import { currentModeState, currentProjectIdState, userSettingsState } from '../recoilModel';
+
+import TelemetryClient from './TelemetryClient';
+
+export const useInitializeLogger = () => {
+  const projectId = useRecoilValue(currentProjectIdState);
+  const page = useRecoilValue(currentModeState);
+  const { telemetry } = useRecoilValue(userSettingsState);
+
+  useEffect(() => {
+    TelemetryClient.setup(telemetry, () => ({
```

Comment:
This can probably just move outside of the useEffect since it is not costly and will ensure the client is setup before the next render phase.